src/main/java/org/zalando/stups/tokens/ClientCredentials.java

```diff
@@ -15,7 +15,28 @@
  */
 package org.zalando.stups.tokens;
 
+/**
+ * "Model" interface specification for client credentials (as described on the OAuth2.0 spec).
+ *
+ * This might be any implementation from a static class that will provide the same <i>id</i> and
+ * <i>secret</i> for all calls up to a {@link ThreadLocal} based implementation that can supply
+ * different values for each {@link Thread}.
+ */
 public interface ClientCredentials {
+
+    /**
+     * Get the <i>id</i> for this {@link ClientCredentials}; this is called the <i>client_id</i> on
```

Comment:
But that be an API level change. Should involve a deprecation period and once renamed a major version bump, right?